homeassistant/components/tesla/switch.py

```diff
@@ -19,6 +19,8 @@
             entities.append(UpdateSwitch(device, controller, config_entry))
         elif device.type == "maxrange switch":
             entities.append(RangeSwitch(device, controller, config_entry))
+        elif device.type == "sentry mode switch":
```

Comment:
Please check that the car supported sentry mode before adding the switch

```
2020-03-18 15:28:11 ERROR (MainThread) [homeassistant.helpers.entity] Update for switch.xizzy_sentry_mode_switch fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 279, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 474, in async_device_update
    await self.async_update()
  File "/config/custom_components/tesla/switch.py", line 147, in async_update
    await super().async_update()
  File "/config/custom_components/tesla/__init__.py", line 258, in async_update
    await self.tesla_device.async_update()
  File "/usr/local/lib/python3.7/site-packages/teslajsonpy/sentry_mode.py", line 46, in async_update
    self.__sentry_mode = data and data["sentry_mode"]
KeyError: 'sentry_mode'
```

This breaks for v1.0 cars

Reply:
Probably can catch the key error here and then not append.

Reply:
Alternatively, within the teslajsonpy probably could have the entity catch the `KeyError` then set `entity_registry_enabled_default` to False.

Reply:
In teslajsonpy, the switch is added during initialization (`get_state()`), regardless of compatibility, and before actually knowing if car is sentry-enabled (`sentry_mode_available` attribute in `vehicle_state`).

One option would be to add a `is_sentry_mode_available()` method to the library. If `false`, do not append the `SentryModeSwitch`. This would require a teslajsonpy update.

Reply:
The library should either handle this and expose a way to know what features are supported or raise a library specific error. We should not have to catch `KeyError` here.

Reply:
The library, since 0.6.0, will set the switch as unavailable if sentry mode is not supported.

Reply:
Confirmed its showing unavailable for an older tesla, and works for a newer one